### DISCUSSION TITLE
New parser for parsing the output of `git diff --name-status`

### DISCUSSION THREAD

**Discussion by kishaningithub** [kellyjonbrazil/jc#397] [Ideas]
The `git diff --name-status main..<<branch>>` gives the output in the format

```
R100    README.md       README_now.md
M       bootstrap/bootstrap.sh
D       configs/sample.json
A       scripts/status.py
```

if you look at above the command gives output which says

- The type of change in the file viz Added, Modified, Deleted, Renamed (there are also ones like file type changed etc)
- The full file path

It would be great to have this outputted as a json

**Comment by kellyjonbrazil** [comment]
This could be a good parser. From the `git-diff` man page it looks like the possible types are:
```
files that are Added (A), Copied (C), Deleted (D), Modified (M), Renamed
(R), have their type (i.e. regular file, symlink, submodule, ...) changed (T), are
Unmerged (U), are Unknown (X), or have had their pairing Broken (B)
```
A schema could look something like:
```
{
  "added": [
    "filename1", "filename2"
  ],
  "deleted": [
    "filename1", "filename2"
  ]
  ... etc.
}
```
